MedallionShell/Shell.cs

```diff
@@ -76,6 +83,24 @@ public Command Run(string executable, IEnumerable<object>? arguments = null, Act
             return command;
         }
 
+        // internal for testing
+        internal static string? GetFullPathUsingSystemPathOrDefault(string executable)
+        {
+            var paths = Environment.GetEnvironmentVariable("PATH")!.Split(Path.PathSeparator);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                var pathExtensions = Environment.GetEnvironmentVariable("PATHEXT")!
+                    .Split(Path.PathSeparator)
+                    .Concat(new[] { string.Empty })
```

Comment:
The user may have included the extension, in which case we shouldn't be adding more.

Given that there may be an executable like `myExe.exe.exe`, we shouldn't be checking whether an extension already exists in the `executable` variable.

Reply:
What is the OS precedence behavior here?

E.g. if I type `dotnet.exe` and there is a file `dotnet.exe` and `dotnet.exe.exe` on my path, which one gets executed?

What if `dotnet.exe.exe` is in a folder which is higher precedence?

What if there is a file `dotnet` and `dotnet.exe`?

We should strive to match the OS behavior exactly, documenting and testing each decision point.